dashboard/app/models/stage.rb

```diff
@@ -152,4 +151,9 @@ def lockable_state(students)
       }
     end
   end
+
+  # Ensures we get the cached ScriptLevels, vs hitting the db.
+  def cached_script_levels
+    script_levels.map{|sl| Script.cache_find_script_level(sl.id)}
```

Comment:
Though the two objects `sl` and `Script.cache_find_script_level(sl.id)` are both `Script` AR objects, the former object is a shallow record that would require an extra query to retrieve any associations, while the latter object already has `{levels: [:game, :concepts] }, :stage, :callouts` associations eager-loaded due to the `#includes` statement originally used to create the object.